### DISCUSSION TITLE
Invoking Scriptkit defocuses inputs, making clipboard actions fail

### DISCUSSION THREAD

**Discussion by pxlprfct** [johnlindquist/kit#840] [Errors] [answered]
I think I might have a setting wrong or something - but currently pasting from Scriptkit doesn't work for me.

## What happens
1. I click on an input
2. I open Scriptkit (The input has now been deselected)
4. I find my emoji
5. I select it
6. ScriptKit _tries_ add it to the page, but fails as the input is no longer selected.

https://user-images.githubusercontent.com/3811773/179402719-24239e94-5d34-4332-81bb-28276e7ef600.mp4

## What should happen
1. I click on an input
2. I open Scriptkit (The input has not been deselected)
4. I find my emoji
5. I select it
6. ScriptKit adds it to the input sucessfully.

I've attached how it works in Alfred as an example of how they make it work (I'm sure it's wizardry behind the scenes)

**Comment by johnlindquist** [comment, marked as answer]
Have you given Kit.app the accessibility access then quit/reopen Kit.app?

It needs permission to perform a "paste" action

**Reply by pxlprfct** [reply]
<img width="780" alt="Screenshot 2022-07-17 at 16 04 38" src="https://user-images.githubusercontent.com/3811773/179404432-df3195cc-d640-4403-a8a9-e258912a0e13.png">

~I'm afraid so! But will do it again to absolutely confirm. I can hear the little error plink in the background after attempting to paste.~

I just checked and rechecked it again (despite it already being checked) and quit the app - and... it works now 🙃🙃🙃🙃🙃🙃 Not quite sure what's was happening there - but very happy it's fixed!

I _did_ reinstall the app previously - so maybe the accessibility checkbox selection was for the old 'version' (I'll be honest I'm not quite sure how accessibility stuff works on OS level).

Thanks so much - this is an amazing bit of _kit_! 😮

Will keep trying, but fairly sure this also fixes https://github.com/johnlindquist/kit/discussions/839 too

**Reply by johnlindquist** [reply]
Awesome, always happy to help. Permissions on osx are always a little flaky and we have zero control other than that permissions UI 🤷‍♂️

If you use Discord, we're starting up a server

https://discord.gg/3jHCQ8P7
